wagtail/blocks/base.py

```diff
@@ -269,6 +273,60 @@ def render(self, value, context=None):
 
         return mark_safe(render_to_string(template, new_context))
 
+    def get_preview_context(self, value, parent_context=None):
+        # We do not fall back to `get_context` here, because the preview context
+        # will be used for a complete view, not just the block. Instead, the
+        # default preview context uses `{% include_block %}`, which will use
+        # `get_context`.
+        return parent_context or {}
+
+    def get_preview_template(self, value=None, context=None):
+        # We do not fall back to `get_template` here, because the template will
+        # be used for a complete view, not just the block. In most cases, the
+        # block's template cannot stand alone for the preview, as it would be
+        # missing the necessary static assets.
+        #
+        # Instead, the default preview template uses `{% include_block %}`,
+        # which will use `get_template` if a template is defined.
+        return (
+            getattr(self.meta, "preview_template", None)
+            or self.DEFAULT_PREVIEW_TEMPLATE
+        )
+
+    def get_preview_value(self):
+        if hasattr(self.meta, "preview_value"):
+            return self.normalize(self.meta.preview_value)
+        return self.get_default()
```

Comment:
We're not sure whether the preview value should fall back to the block's `default` or not. Please let me know if we'd rather not.

Reply:
Seems reasonable enough to fall back on `default` to me.